modules/xcode/xcode_common.lua

```diff
@@ -1363,6 +1363,18 @@
 		end
 	end
 
+	function xcode.XCBuildConfiguration_SwiftLanguageVersion(settings, cfg)
+		-- if no swiftversion is provided, don't set swift version
+		-- Projects with swift files but without swift version will refuse 
+		-- to build on Xcode but setting a default SWIFT_VERSION may have 
+		-- unexpected interactions with other systems like cocoapods
+		if not cfg.swiftversion then
+			return
+		else
```

Comment:
I think it would be simpler if you reverse the check:
```suggestion
		if cfg.swiftversion then
```